src/native/corehost/comhost/comhost.cpp

```diff
@@ -66,8 +98,58 @@ namespace
 
                 return StatusCode::Success;
             },
-            delegate
+            [load_context](pal::dll_t fxr, hostfxr_handle context)
+            {
+                *load_context = ISOLATED_CONTEXT;
+                auto get_runtime_property_value = reinterpret_cast<hostfxr_get_runtime_property_value_fn>(pal::get_symbol(fxr, "hostfxr_get_runtime_property_value"));
+                const pal::char_t* value;
+                if (get_runtime_property_value(context, _X("System.Runtime.InteropServices.COM.LoadComponentInDefaultContext"), &value) == StatusCode::Success
+                    && pal::strcasecmp(value, _X("true")) == 0)
+                {
+                    *load_context = nullptr; // Default context
+                }
+            },
+            reinterpret_cast<void**>(&get_function_pointer)
         );
+        if (status != StatusCode::Success)
+            return status;
+
+        const pal::char_t* method_name;
+        switch (del_type)
+        {
+        case hostfxr_delegate_type::hdt_com_activation:
+            method_name = _X("GetClassFactoryForTypeInContext");
+            break;
+        case hostfxr_delegate_type::hdt_com_register:
+            method_name = _X("RegisterClassForTypeInContext");
+            break;
+        case hostfxr_delegate_type::hdt_com_unregister:
+            method_name = _X("UnregisterClassForTypeInContext");
+            break;
+        default:
+            return StatusCode::InvalidArgFailure;
+        }
+
+        status = get_function_pointer(
+            _X("Internal.Runtime.InteropServices.ComActivator, System.Private.CoreLib"),
+            method_name,
+            UNMANAGEDCALLERSONLY_METHOD,
+            nullptr, // load context
+            nullptr, // reserved
+            (void**)delegate);
+
+        if (status == StatusCode::Success)
+            return status;
+
+        // Newer methods with context not found and using isolated context.
+        // Fall back to methods without context
+        if (status == 0x80131513 /*MissingMethodException*/ && *load_context == ISOLATED_CONTEXT)
```

Comment:
Can we also reference the HRESULT form? I like the managed exception type, but also referencing the common HRESULT is helpful.

Reply:
Well, this is unfortunate. Here is what I have now:
```c++
// The runtime will throw MissingMethodException, so we check for the corresponding COR_E_MISSINGMETHOD HRESULT.
// We also need to check for COR_E_MISSINGMEMBER due to a pre-7.0 bug where the HRESULT was not correctly set on
// MissingMethodException and it ended up with the HRESULT for MissingMemberException
if ((status == 0x80131513 /*COR_E_MISSINGMETHOD*/ || status == 0x80131512 /*COR_E_MISSINGMEMBER*/)
    && *load_context == ISOLATED_CONTEXT)
```